package/scripts/ambari_helpers.py

```
@@ -53,27 +66,31 @@ def has_hive():
   else:
     return false
 
-### TODO: Update this with a safe version, as hdp-select doesn't exist on HDP < 2.2
 def get_hdp_version():
-  try:
-    command = 'hdp-select status hadoop-client'
-    return_code, hdp_output = shell.call(command, timeout=20)
-  except Exception, e:
-    Logger.error(str(e))
-    raise Fail('Unable to execute hdp-select command to retrieve the version.')
+  command = 'hadoop version | head -n 1 | cut -d. -f4-'
```

pipes will mask failure